src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -236,6 +237,22 @@ public GoogleBucketResource lookupBucketMetadata(String bucketResourceId) {
     return bucketService.getBucketResourceById(UUID.fromString(bucketResourceId), false);
   }
 
+  /**
+   * Fetch an existing storage_account metadata row. Note this method does not check for the
+   * existence of the underlying cloud resource. This method is intended for places where an
+   * existence check on the associated cloud resource might be too much overhead (e.g. DRS lookups).
+   * Most storage account lookups should use the lookupStorageAccount method instead, which has
+   * additional overhead but will catch metadata corruption errors sooner.
+   *
+   * @param storageAccountId our identifier for the storage account
+   * @return a reference to the storage account as a POJO AzureStorageAccountResource
+   * @throws AzureResourceNotFoundException if the storage_account metadata row does not exist
+   */
+  public AzureStorageAccountResource lookupStorageAccountMetadata(String storageAccountId) {
+    return storageAccountService.getStorageAccountResourceById(
```

Comment:
Looking more at this class I'm wondering if it should be split into azure and gcp specific classes. There are azure and gcp specific fields, and from a quick read it doesn't look like there's any code that uses both platforms at the same time. Is there common code that would be hard to share if this was split into two classes?

On the other hand, you could move the platform specific code into this class and it could switch internally, instead of putting that code in the platform wrapper as I suggested above. The caller interface would be platform neutral and the platform specific APIs would be hidden inside the service code.

Reply:
Yeah I kinda like that second option.  I may try to do that in a followon.  On the other hand, It's not quite apples to apples between the two cloud usecases so perhaps a splitting into two classes might make sense